clouddriver-sql/src/main/kotlin/com/netflix/spinnaker/clouddriver/sql/security/SqlAccountDefinitionRepository.kt

```diff
@@ -0,0 +1,216 @@
+    withPool(poolName) {
+      jooq.read { ctx ->
+        ctx.select(idColumn)
+          .from(accountsTable)
+          .where(deletedColumn.isTrue, typeColumn.eq(typeName), lastModifiedColumn.ge(startTimestampMillis))
+          .orderBy(lastModifiedColumn)
+          .fetch { (id) -> id }
+      }
+    }
+
+  override fun listByType(typeName: String): MutableList<out CredentialsDefinition> =
+    withPool(poolName) {
+      jooq.read { ctx ->
+        ctx.select(bodyColumn)
+          .from(accountsTable)
+          .where(deletedColumn.isFalse, typeColumn.eq(typeName))
+          .fetch { (json) ->
+            mapper.convertFromString(json.data(), typeName)
+          }
+      }
+    }
+
+  override fun create(definition: CredentialsDefinition) {
+    withPool(poolName) {
+      jooq.transactional { ctx ->
+        val timestamp = clock.millis()
+        val user = AuthenticatedRequest.getSpinnakerUser().orElse("anonymous")
+        val body = JSON.valueOf(mapper.convertToString(definition))
+        val typeName = AccountDefinitionMapper.getJsonTypeName(definition.javaClass)
+        ctx.insertInto(accountsTable)
+          .set(idColumn, definition.name)
+          .set(typeColumn, typeName)
+          .set(bodyColumn, body)
+          .set(createdColumn, timestamp)
+          .set(lastModifiedColumn, timestamp)
+          .set(modifiedByColumn, user)
+          .execute()
+        ctx.insertInto(accountHistoryTable)
+          .set(idColumn, definition.name)
+          .set(typeColumn, typeName)
+          .set(bodyColumn, body)
+          .set(lastModifiedColumn, timestamp)
+          .set(versionColumn, 1)
+          .execute()
+      }
+    }
+  }
+
+  override fun update(definition: CredentialsDefinition) {
+    withPool(poolName) {
+      jooq.transactional { ctx ->
+        val timestamp = clock.millis()
+        val user = AuthenticatedRequest.getSpinnakerUser().orElse("anonymous")
+        val body = JSON.valueOf(mapper.convertToString(definition))
+        ctx.update(accountsTable)
+          .set(bodyColumn, body)
+          .set(lastModifiedColumn, timestamp)
+          .set(modifiedByColumn, user)
+          .set(deletedColumn, false)
+          .where(idColumn.eq(definition.name))
+          .execute()
+        val version = ctx.select(versionColumn + 1)
+          .from(accountHistoryTable)
+          .where(idColumn.eq(definition.name))
+          .orderBy(versionColumn.desc())
+          .limit(1)
+        ctx.insertInto(accountHistoryTable)
+          .set(idColumn, definition.name)
+          .set(typeColumn, AccountDefinitionMapper.getJsonTypeName(definition.javaClass))
+          .set(bodyColumn, body)
+          .set(lastModifiedColumn, timestamp)
+          .set(versionColumn, version)
+          .execute()
+      }
+    }
+  }
+
+  override fun delete(name: String) {
+    withPool(poolName) {
+      jooq.transactional { ctx ->
+        ctx.update(accountsTable)
```

Comment:
- [ ] Change deletion behavior to use `DELETE FROM` rather than updating a deletion flag.